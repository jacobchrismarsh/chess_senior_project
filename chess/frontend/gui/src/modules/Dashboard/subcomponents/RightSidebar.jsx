import React, { Component } from "react";

export class RightSidebar extends React.Component { 
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div>
        <br/>
        Current games
      </div>
    );
  }
}