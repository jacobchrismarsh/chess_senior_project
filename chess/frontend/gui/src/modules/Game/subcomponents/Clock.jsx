import React from "react";
import Countdown from "react-countdown-now"
import SweetAlert from "sweetalert2-react";
import "../clock.css"

export class Clock extends React.Component {
  constructor(props) {
    super(props);
    this.clock = React.createRef();
  }

  componentWillUpdate() {
    let { move, turn } = this.props
    let clockApiInterface = this.clock.current.api;

    if ((this.state.playerColor === turn) && move !== 0) {
      clockApiInterface.start();
    } else {
      clockApiInterface.pause();
    }

  }

  render() {
    return (
      <Countdown 
        ref={this.clock}
        autoStart={false}
        daysInHours={true}
        date={this.props.time} />
    );
  }
}