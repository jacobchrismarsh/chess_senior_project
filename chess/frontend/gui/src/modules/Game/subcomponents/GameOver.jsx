import React from "react";
import { WHITE, BLACK } from "../constants";
import "../piecepromotion.css"; // same stuff here so might as well

export class GameOver extends React.Component {
  render() {
    let { show, winner } = this.props;

    if (!show) {
      return null;
    }

    return (
      <div className="overlay">
        <div className="overlay-container game-over-text">
          Game Over! { winner } wins!
        </div>
      </div>
    );
  }
}
