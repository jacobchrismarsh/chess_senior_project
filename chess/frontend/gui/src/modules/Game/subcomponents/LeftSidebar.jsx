import React from "react";
import "../game.css";

export class LeftSidebar extends React.Component {
  renderCapturedWhitePieces() {
    const list = this.props.white.map((piece, index) => {
      return (
        <li key={index}>
          <img src={piece.svg} width="25" height="25" />
        </li>
      );
    });

    return <ul>{list}</ul>;
  }

  renderCapturedBlackPieces() {
    const list = this.props.black.map((piece, index) => {
      return (
        <li key={index}>
          <img src={piece.svg} width="25" height="25" />
        </li>
      );
    });

    return <ul>{list}</ul>;
  }

  render() {
    return (
      <div className="sidebar">
        <div className="left-sidebar-container">
          <div className="left-sidebar-item color-black">
            {this.renderCapturedWhitePieces()}
          </div>
          <div className="left-sidebar-item">
            {this.renderCapturedBlackPieces()}
          </div>
        </div>
      </div>
    );
  }
}
