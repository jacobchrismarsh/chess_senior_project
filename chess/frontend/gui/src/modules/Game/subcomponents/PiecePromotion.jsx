import React from "react";
import { Square } from "./Square";
import { WHITE } from "../constants";

import blackQueen from "../../../cburnett/bq.svg"
import whiteQueen from "../../../cburnett/wq.svg"
import blackRook from "../../../cburnett/br.svg"
import whiteRook from "../../../cburnett/wr.svg"
import blackKnight from "../../../cburnett/bn.svg"
import whiteKnight from "../../../cburnett/wn.svg"
import blackBishop from "../../../cburnett/bb.svg"
import whiteBishop from "../../../cburnett/wb.svg"
import "../piecepromotion.css";

export class PiecePromotion extends React.Component {

  constructor(props) {
    super(props);
  }

  render() {
    let { color, show, handleClick } = this.props;
    let whitePieces = [whiteQueen, whiteRook, whiteKnight, whiteBishop];
    let blackPieces = [blackQueen, blackRook, blackKnight, blackBishop];
    let usePieces = (color === WHITE ? whitePieces :  blackPieces);

    if (!show) {
      return null;
    }

    return (
      <div className="overlay">
        <br/>
        <div>Select the piece you want to promote to!</div>
        <div className="overlay-container">
          {
            usePieces.map((piece, i) => {
              return (
                <Square
                  key={i}
                  piece={piece}
                  selected={null}
                  movable={null}
                  squareColor={null}
                  onClick={() => handleClick(i)}
                />
              );
            })
          }
        </div>
      </div>
    );
  }
}
