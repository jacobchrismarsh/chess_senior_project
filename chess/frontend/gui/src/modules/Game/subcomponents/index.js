export * from "./Board";
export * from "./Square";
export * from "./RightSidebar";
export * from "./LeftSidebar";
export * from "./Clock";
export * from "./PiecePromotion";
