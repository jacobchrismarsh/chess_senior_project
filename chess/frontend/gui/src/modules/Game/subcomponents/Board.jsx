import React from "react";
import { Square } from "./Square";
import { Clock } from "./Clock";
import { WHITE, BLACK } from '../constants';
import "../board.css";

const START_MIN = 10;
const START_SECONDS = 0;

export class Board extends React.Component {
  renderSquare(i, squareColor) {
    return (
      <Square
        key={i}
        piece={this.props.squares[i] ? this.props.squares[i].svg : null}
        selected={this.props.squares[i] ? this.props.squares[i].selected : null}
        movable={this.props.squares[i] ? this.props.squares[i].movable : null}
        squareColor={squareColor}
        onClick={() => this.props.handleClick(i)}
      />
    );
  }

  isEven(num) {
    return num % 2 === 0;
  }

  getSquareColor(i, j) {
    return (this.isEven(i) && this.isEven(j)) ||
      (!this.isEven(i) && !this.isEven(j))
      ? "light"
      : "dark";
  }

  render() {
    const board = [];
    for (let i = 0; i < 8; i++) {
      const squareRows = [];
      for (let j = 0; j < 8; j++) {
        squareRows.push(
          this.renderSquare(i * 8 + j, this.getSquareColor(i, j))
        );
      }
      board.push(
        <div key={i} className="board-row">
          {squareRows}
        </div>
      );
    }
    return (
      <div className="tri-list">
        <Clock
          turn={this.props.turn}
          minutes={START_MIN}
          seconds={START_SECONDS}
          player={BLACK}
          move={this.props.move}
        />
        <br />
        {board}
        <br />
        <Clock
          turn={this.props.turn}
          minutes={START_MIN}
          seconds={START_SECONDS}
          player={WHITE}
          move={this.props.move}
        />
      </div>
    );
  }
}
