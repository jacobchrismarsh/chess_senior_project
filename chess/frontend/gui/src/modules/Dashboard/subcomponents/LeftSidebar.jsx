import React, { Component } from "react";
import RadioButtonGroup from '@material-ui/core/RadioGroup';
import RadioButton from '@material-ui/core/Radio';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Button from '@material-ui/core/Button';
import { Slider } from 'material-ui-slider';
import autoBind from 'react-autobind';
import TextField from '@material-ui/core/TextField';
import { WHITE, BLACK } from "../../Game/constants"
import $ from 'jquery';

import "../dashboard.css";

export class LeftSidebar extends React.Component { 
  constructor(props) {
    super(props);

    this.state = {
      color: WHITE,
      opponent: 'Computer',
      opponentUsername: '',
      difficulty: 10
    };

    autoBind(this);
  }

  handleChangeRadioButton(component, event) {
    return e => {
      let value = e.target.value
      this.setState({
        [component]: value
      })
    }
  }

  handleChangeSlider(component, event) {
    return e => {
      this.setState({
        [component]: e
      });
    }
  }

  handleChangeText(component, event) {
    return e => {
      let value = e.target.value;
      this.setState({
        [component]: value
      });
    }
  }

  handleButton() {
    return $.ajax({
      url: "http://127.0.0.1:8000/game/create/",
      method: "GET",
      date: {
        opponent: this.state.opponent,
        difficulty: this.state.difficulty,
        color: this.state.color,
        opponentUsername: this.state.opponentUsername
      }
    });
  }

  render() {
    let colorOptions = [WHITE, BLACK]
    let opponentOptions = ['Computer', 'Online Opponent'];

    return (
      <div className="sidebar-container">

        <div className="dashboard-left-sidebar-item">
          Start a new game
        </div>

        <div className="dashboard-left-sidebar-item">
          <div>What color would you like to be?</div>
          <br/>
          <RadioButtonGroup
            value={this.state.color}
            onChange={this.handleChangeRadioButton('color')}
          >

          {
            colorOptions.map((choice, idx) =>
              <FormControlLabel
                key={idx}
                value={choice}
                control={<RadioButton />}
                label={choice}
              />)
          }
          </RadioButtonGroup>
        </div>

        <div className="dashboard-left-sidebar-item">
          <div>Who would you like to play against?</div>
          <br/>
          <RadioButtonGroup
            value={this.state.opponent}
            onChange={this.handleChangeRadioButton('opponent')}
            
          >

          {
            opponentOptions.map((choice, idx) =>
              <FormControlLabel
                key={idx}
                value={choice}
                control={<RadioButton />}
                label={choice}
              />)
          }
          </RadioButtonGroup>
        </div>
        {
          this.state.opponent === 'Computer' && (
            <div className="dashboard-left-sidebar-item slider-div">
              <div>Difficuly</div>
              <br/>
              <Slider
                onChange={this.handleChangeSlider('difficulty')}
                min={0}
                max={20}
                defaultValue={10}
                />
            </div>
          )
        }

        {
          this.state.opponent === 'Online Opponent' && (
            <div className="dashboard-left-sidebar-item dashboard-text-input">
            <div>Opponent Username</div>
            <br/>
            <TextField
              value={this.state.opponentUsername}
              onChange={this.handleChangeText('opponentUsername')}
            />
            </div>
          )
        }

        <div className="dashboard-left-sidebar-item">
          <Button
            size="large"
            variant="contained"
            color="primary"
            className="chess-button"
            onClick={this.handleButton}
          >
            Create Game
          </Button>
        </div>

      </div>
    );
  }
}