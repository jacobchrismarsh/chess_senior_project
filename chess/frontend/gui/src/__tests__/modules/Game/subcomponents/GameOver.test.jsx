import React from "react";
import { shallow } from 'enzyme';
import { GameOver } from '../../../../modules/Game/subcomponents/GameOver';

describe("modules/Game/subcomponents/GameOver.jsx", () => {
  it("should render the GameOver components", () => {
    let gameOver = shallow(<GameOver show={true} winner="Yeet" />);
    expect(gameOver).toMatchSnapshot();
  });

  it("should not render the GameOver components", () => {
    let gameOver = shallow(<GameOver show={false} winner="Yeet" />);
    expect(gameOver).toMatchSnapshot();
  });
});