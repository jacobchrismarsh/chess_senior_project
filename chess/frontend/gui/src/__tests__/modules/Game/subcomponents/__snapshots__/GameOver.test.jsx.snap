// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/Game/subcomponents/GameOver.jsx should not render the GameOver components 1`] = `""`;

exports[`modules/Game/subcomponents/GameOver.jsx should render the GameOver components 1`] = `
<div
  className="overlay"
>
  <div
    className="overlay-container"
  >
    Game Over! 
    Yeet
     wins!
  </div>
</div>
`;
